CMakeLists.txt

```diff
@@ -253,20 +253,43 @@ if (NOT WIN32)
 endif()
 
 if(SYSTEM_BOOST)
-  set(Boost_USE_MULTITHREADED      ON)
-  find_package( Boost 1.50.0 COMPONENTS  thread system filesystem program_options regex test_exec_monitor )
-endif()
-
-if (Boost_FOUND)
-	link_directories(${Boost_LIBRARY_DIRS})
+    set(Boost_USE_MULTITHREADED ON)
+    find_package( Boost 1.50.0 COMPONENTS  thread system filesystem program_options regex test_exec_monitor )
+    if (Boost_FOUND)
+        # we're specifically setting up the boost libraries here, so we can
+        # name them
+        add_library(boost_thread_lib SHARED IMPORTED)
+        set_property(TARGET boost_thread_lib PROPERTY IMPORTED_LOCATION
+          ${Boost_THREAD_LIBRARY})
+        add_library(boost_system_lib SHARED IMPORTED)
+        set_property(TARGET boost_system_lib PROPERTY IMPORTED_LOCATION
+          ${Boost_SYSTEM_LIBRARY})
+        add_library(boost_filesystem_lib SHARED IMPORTED)
+        set_property(TARGET boost_filesystem_lib PROPERTY IMPORTED_LOCATION
+          ${Boost_FILESYSTEM_LIBRARY})
+        add_library(boost_program_options_lib SHARED IMPORTED)
+        set_property(TARGET boost_program_options_lib PROPERTY
+          IMPORTED_LOCATION ${Boost_PROGRAM_OPTIONS_LIBRARY})
+        add_library(boost_regex_lib SHARED IMPORTED)
+        set_property(TARGET boost_regex_lib PROPERTY IMPORTED_LOCATION
+          ${Boost_REGEX_LIBRARY})
+        add_library(boost_test_exec_monitor_lib SHARED IMPORTED)
+        set_property(TARGET boost_test_exec_monitor_lib PROPERTY
+          IMPORTED_LOCATION ${Boost_TEST_EXEC_MONITOR_LIBRARY})
+        set(boost_include_dirs ${Boost_INCLUDE_DIRS})
+    else()
+        set(SYSTEM_BOOST OFF)
+        set(boost_include_dirs ${CMAKE_SOURCE_DIR}/external_libraries/boost)
+        message(WARNING "Not using system boost because none was found.")
```

Comment:
i think this should be a FATAL_ERROR; i can't think of a situation where this is what the user actually wants to happen

Reply:
You're probably right. I thought this would be a nice "fallback", but explicitely failing is better, indeed.